### PR TITLE
Additional guidance for contributions

### PR DESCRIPTION
**Pull request by wchao1115**
Following the WG teleconference call on 4/13 on the topic of contributions guidelines, I've added more text into our existing contributing.md file to further clarify the types of change and provide additional guidance on how to work on a PR and what to expect from the working group's review process. Most of this change is what I verbally outlined on the call that day with additional clarification. Further feedback here is welcome. 

BTW, I of course realize that this PR itself doesn't follow the guidelines it conveys in a strict sense, but the fact that this PR does follow an agreement from our WG conference should give it a pass. ^^

### CODE REVIEW

File: CONTRIBUTING.md
Comment (anssiko) [inline CONTRIBUTING.md:7]: I personally like the category names introduced in this PR because they're more generic and as such familiar to a broader group of OSS contributors. We could consider mapping these to the W3C Process defined [classes of changes](https://www.w3.org/2021/Process-20211102/#correction-classes) that are more W3C specific and introduce both the W3C Process and Patent Policy linkage. Another possible W3C Process alignment is to speak about [substantive change](https://www.w3.org/2021/Process-20211102/#substantive-change) rather than material impact. Established W3C terminology is to speak of normative vs. informative or non-normative content (W3C Process does not have formal definitions for these terms so the generic meaning is assumed).

Using W3C parlance in this guideline doc might be too much so I'm not pushing that strongly. @dontcallmedom WDYT?

File: CONTRIBUTING.md
Comment (anssiko) [inline CONTRIBUTING.md:24]: s/Machine-Learning/Machine Learning

I'd drop "and get a vote" because [deciding by voting](https://www.w3.org/2021/Process-20211102/#Votes) has a special meaning in W3C and we're not doing that for technical work.

File: CONTRIBUTING.md
Comment (anssiko) [inline CONTRIBUTING.md:29]: We may want to add (not necessarily in these exact words):

A pull request can be merged when it has received adequate review from the Working Group. The editors and the chair are responsible for ensuring review is sought from appropriate people. For substantive changes, adequate review is two or more reviewers. The bi-weekly teleconference calls provide another venue for providing feedback and review open pull requests.

**Review by anssiko — APPROVED** [review]
@wchao1115 much thanks for the PR!

I believe this PR, and specifically its guidelines for _new content_, will fix our long-standing issue #231 :-)

File: CONTRIBUTING.md
Comment (huningxin) [inline CONTRIBUTING.md:27]: For new operation proposal, should we define a template / check list for contributors to fill? 

The existing [Guidelines for new operations](https://www.w3.org/TR/webnn/#security-new-ops)  would be a good check list from security perspective. Based on the discussion of [Rationale/criteria for adding new ops to the WebNN API](https://www.w3.org/2021/11/18-webmachinelearning-minutes.html#t02), some additional aspects may need to be included in the github issue, e.g.:
- use case
- sample model
- cross-frameworks support
- cross-platforms implementability

File: CONTRIBUTING.md
Comment (wchao1115) [reply CONTRIBUTING.md:24]: Good to know. Will fix that.

File: CONTRIBUTING.md
Comment (wchao1115) [reply CONTRIBUTING.md:27]: Please check out the new commit.

Comment (huningxin) [reply CONTRIBUTING.md:27]: LGTM, thanks much!

**Review by huningxin — APPROVED** [review]
LGTM. This is a great guideline. Thanks much @wchao1115 !

**Comment by zolkis** [comment]
From the description it is not clear whether e.g. adding a missing algorithm is new content or wording change. 
Actually, much of the PRs for adding an algorithm will contain new content, wording change and might contain stylistic adaptations as well. 
A long-term learning over my career has been that classifying the world does not necessarily help with simplicity, since categories will always be diffuse. One should seek classification of operations instead. For instance, my PRs will actually be "adding algorithm to op", and that is more clear/simple than saying "a new PR that contains new content, wording change and stylistic changes", or, splitting it into 3 PRs.

**Comment by zolkis** [comment]
Another question emerges: how a PR is supposed to be flagged for one or more of the categories?
Currently labeling is restricted, so the only possibility is to mention it in words in the PR itself.

**Pull request merged**
